snuba/querylog/__init__.py

```diff
@@ -115,32 +117,43 @@ def _add_tags(
                     break
 
 
-def record_invalid_request(timer: Timer, referrer: Optional[str]) -> None:
+def record_invalid_request(
+    timer: Timer, new_status: NewStatus, referrer: Optional[str]
+) -> None:
     """
     Records a failed request before the request object is created, so
     it records failures during parsing/validation.
     This is for client errors.
     """
-    _record_failure_building_request(QueryStatus.INVALID_REQUEST, timer, referrer)
+    _record_failure_building_request(
+        QueryStatus.INVALID_REQUEST, new_status, timer, referrer
+    )
 
 
-def record_error_building_request(timer: Timer, referrer: Optional[str]) -> None:
+def record_error_building_request(
+    timer: Timer, new_status: NewStatus, referrer: Optional[str]
+) -> None:
     """
     Records a failed request before the request object is created, so
     it records failures during parsing/validation.
     This is for system errors during parsing/validation.
     """
-    _record_failure_building_request(QueryStatus.ERROR, timer, referrer)
+    _record_failure_building_request(QueryStatus.ERROR, new_status, timer, referrer)
 
 
 def _record_failure_building_request(
-    status: QueryStatus, timer: Timer, referrer: Optional[str]
+    status: QueryStatus, new_status: NewStatus, timer: Timer, referrer: Optional[str]
 ) -> None:
     # TODO: Revisit if recording some data for these queries in the querylog
     # table would be useful.
     if settings.RECORD_QUERIES:
         timer.send_metrics_to(
             metrics,
-            tags={"status": status.value, "referrer": referrer or "none"},
+            tags={
+                "status": status.value,
+                "referrer": referrer or "none",
+                "new_status": new_status.status.value,
+                "slo": new_status.slo.value,
```

Comment:
This is likely going to be a problem with our metrics cardinality, thus cost.
These tags are attached to all the marks in a query (every phase of the query has a metrics and each one of them will have all these tags). 
Each mark is then multiplied by Veneur. This is likely to multiply the snuba metrics 4 times and Snuba is the main offender today. 
I would suggest against applying the new status to all the marks of the query. This will probably require touching the Timer class.

Reply:
I don't think this gets sent for every mark. There is a separate `mark_tags` that gets sent for the marks, and this is separate from that. https://github.com/getsentry/snuba/blob/master/snuba/utils/metrics/timer.py#L92-L94

So I don't think I need to make a change there.